vancouver-drsm.csl

```diff
@@ -0,0 +1,466 @@
+  </macro>
+  <macro name="pages">
+    <choose>
+      <if type="article-journal article-magazine article-newspaper review review-book" match="any">
+        <text variable="page" form="short" prefix=":"/>
+      </if>
+      <else-if type="book" match="any">
+        <text variable="number-of-pages" prefix=" "/>
+        <choose>
+          <if is-numeric="number-of-pages">
+            <label variable="number-of-pages" form="short" prefix=" " plural="never"/>
+          </if>
+        </choose>
+      </else-if>
+      <else-if type="entry-dictionary entry-encyclopedia" match="any">
+        <text variable="page" suffix=" p."/>
+      </else-if>
+      <else-if type="legislation" match="any">
+        <choose>
+          <if match="any" variable="page">
+            <text variable="page" suffix="."/>
+          </if>
+          <else>
+            <text value="" suffix="."/>
+          </else>
+        </choose>
+      </else-if>
+      <else>
+        <group prefix=" " delimiter=" ">
+          <label plural="never" variable="page" form="short"/>
+          <text variable="page"/>
+        </group>
+      </else>
+    </choose>
+  </macro>
+  <macro name="localisation-revue">
+    <choose>
+      <if type="article-journal article-magazine review review-book" match="any">
+        <text variable="volume"/>
+        <text variable="issue" prefix="(" suffix=")"/>
+      </if>
+    </choose>
+  </macro>
+  <macro name="details-rapport">
+    <choose>
+      <if type="report">
+        <text variable="number" prefix="Rapport no "/>
+      </if>
+    </choose>
+  </macro>
+  <macro name="date-acces">
+    <group prefix="[" suffix="]">
+      <text term="cited" text-case="capitalize-first" suffix=" "/>
+      <date form="text" variable="accessed"/>
+    </group>
+  </macro>
+  <citation>
+    <sort>
+      <key variable="citation-number"/>
+    </sort>
+    <layout delimiter="," prefix="(" suffix=")">
+      <text variable="citation-number"/>
+    </layout>
+  </citation>
+  <bibliography et-al-min="6" et-al-use-first="6" second-field-align="flush">
+    <sort>
+      <key variable="citation-number"/>
+    </sort>
+    <layout>
+      <text variable="citation-number" suffix=". "/>
+      <choose>
+        <if type="article article-journal article-magazine article-newspaper" match="any">
+          <group suffix=". ">
+            <text macro="auteur"/>
+          </group>
+          <group suffix=". ">
+            <text macro="titre"/>
+          </group>
+          <group suffix=" ">
+            <text macro="titre-contenant"/>
+          </group>
+          <group>
+            <text macro="date"/>
+          </group>
+          <group suffix=":">
+            <text macro="localisation-revue"/>
+          </group>
+          <group suffix=".">
+            <text macro="pages"/>
+          </group>
+        </if>
+        <else-if type="book" match="any">
+          <group suffix=". ">
+            <text macro="auteur"/>
+          </group>
```

Comment:
Did you edit this style by hand? Creating `<group/>` elements with only one child has no benefit, and it's simpler to e.g. just use `<text macro="auteur" suffix=". "/>` instead of

```
<group suffix=". ">
  <text macro="auteur"/>
</group>
```

There are good reasons to use groups, but here you just complicate the style.